src/Symfony/Component/Serializer/Encoder/XmlEncoder.php

```diff
@@ -76,7 +76,12 @@ public function encode($data, string $format, array $context = [])
         $encoderIgnoredNodeTypes = $context[self::ENCODER_IGNORED_NODE_TYPES] ?? $this->defaultContext[self::ENCODER_IGNORED_NODE_TYPES];
         $ignorePiNode = \in_array(\XML_PI_NODE, $encoderIgnoredNodeTypes, true);
         if ($data instanceof \DOMDocument) {
-            return $data->saveXML($ignorePiNode ? $data->documentElement : null);
+            set_error_handler(static function ($type, $message) { throw new NotEncodableValueException($message); });
```

Comment:
be careful. This is currently also turning deprecations into a NotEncodableValueException, which is not OK.

Reply:
I changed to 
```
set_error_handler(static function ($type, $message) { throw new NotEncodableValueException($message); }, \E_ERROR | \E_WARNING);
```